codex/sales/states/errored.nim

```diff
@@ -30,5 +30,5 @@ method run*(state: SaleErrored, machine: Machine): Future[?State] {.async.} =
     onClear(request, data.slotIndex)
 
   if onCleanUp =? agent.onCleanUp:
-    await onCleanUp(returnBytes = true)
```

Comment:
A could have happened in any of the states, eg downloading.

Reply:
Thinking about this again, if an error occurred in preparing, downloading, or filling, then it stands to reason that the node should take another stab at filling the slot. If an occurred in filled or proving, then the node should not try to reprocess the slot. Will update to reflect this.

Reply:
I updated it so that only when `SaleErrored` is reached from `SaleDownloading` will the slot be reprocessed. `SalePreparing`. I was thinking that perhaps we could extend this same behaviour to SalePreparing and SaleInitialProving, however I'm not entirely sure if we would want to reprocess a slot if there's an exception while preparing (availability checking and reservation creation), or initial proving (generating the proof).

Reply:
Nice!
Yeah, lets discuss this on call today. I would maybe incline toward reprocessing for Preparing but not InitialProving?